predict.py

```diff
@@ -166,8 +206,39 @@ def predict(
             n_tokens += 1
             yield decoded_token
         t = time.time() - st
-
+        print(f"hostname: {socket.gethostname()}")
         if debug:
             print(f"cur memory: {torch.cuda.memory_allocated()}")
             print(f"max allocated: {torch.cuda.max_memory_allocated()}")
             print(f"peak memory: {torch.cuda.max_memory_reserved()}")
+
+    # def remove(f: "Callable", defaults: "dict[str, Any]") -> "Callable":
+    #     # pylint: disable=no-self-argument
+    #     # for the purposes of inspect.signature as used by predictor.get_input_type,
+    #     # remove the argument (system_prompt)
+    #     wrapped = functools.partialmethod(f, **defaults)
+    #     sig = inspect.signature(wrapped)
+    #     # TypeError: functools.partialmethod(<function Predictor.predict at 0x7fa5d2136340>, , system_prompt=None) is not a callabl object
+    #
+    #     params = [p for name, p in sig.parameters.items() if name not in defaults]
+    #     wrapped.__signature__ = sig.replace(parameters=params)
+    #     return wrapped
+
+    # if not USE_SYSTEM_PROMPT:
+    #     predict = remove(predict, {"system_prompt": None})
```

Comment:
remove?

Reply:
I'm not sure if I'll want this in the future

If codellama will need to have arguments that are not in the base predict, or there are arguments that aren't applicable to codellama, it would be nice to have a generic way of removing it from the signature

The problem is just that partialmethod doesn't work with inspect.signature for some reason